src/evaluator/functions.ts

```diff
@@ -392,6 +393,31 @@ array['unique'] = async function (args, scope, execute) {
 }
 array['unique'].arity = 1
 
+array['intersects'] = async function (args, scope, execute) {
+  // Intersects returns true if the two arrays have at least one element in common. Only
+  // primitives are supported; non-primitives are ignored.
+  const arr1 = await execute(args[0], scope)
+  if (!arr1.isArray()) {
+    return NULL_VALUE
+  }
+
+  const arr2 = await execute(args[1], scope)
+  if (!arr2.isArray()) {
+    return NULL_VALUE
+  }
+
+  for (const v1 of await arr1.get()) {
+    for (const v2 of await arr2.get()) {
```

Comment:
In other places we use `for await (const elem of arr)`, should we follow that pattern?

Reply:
Sure.